exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -1086,3 +1075,34 @@ interface IDateColumnType {
 interface JsonColumnMarker {
     val usesBinaryFormat: Boolean
 }
+
+/**
+ * Returns the [ColumnType] commonly associated with storing values of type [T], or the [defaultType] if a mapping
+ * does not exist for type [T].
+ *
+ * @throws IllegalStateException If no column type mapping is found and a [defaultType] is not provided.
+ */
+fun <T : Any> resolveColumnType(
```

Comment:
we may want to mark it as internal API annotation

Reply:
@e5l I agree. I originally extracted the function logic from inside the json module, so I left it public to still be accessible there. I'm happy to mark it as internal in the core module (since it's used a lot there) and put back the block in the json function (used only in 1 place).